R/plots.R

```diff
@@ -146,38 +148,39 @@ generate_individual_outputs <- function(data,
   validate_input_data_generate_individual_outputs(
     data = data,
     matched_prioritized = matched_prioritized,
-    target_type = target_type
+    target_type = target_type,
+    by_group = by_group
   )
 
   # create sub directory for the selected institute
-  dir.create(file.path(output_directory, group_id), showWarnings = FALSE)
+  dir.create(file.path(output_directory, by_group_value), showWarnings = FALSE)
 
   data <- data %>%
     dplyr::filter(
-      group_id == .env$group_id,
-      scenario_source == .env$scenario_source,
-      region == .env$region,
-      sector %in% .env$sector
+      .data[[by_group]] == .env$by_group_value,
+      .data[["scenario_source"]] == .env$scenario_source,
+      .data[["region"]] == .env$region,
+      .data[["sector"]] %in% .env$sector
```

Comment:
nit: e.g. `.env$by_group_value` can also be transformed to `.env[[by_group_value]]`, and with `by_group` and `by_group_value` objects floating around, this might be a place where it actually makes a difference (e.g. `.env$by_group` will match `.env$by_group_value` if `by_group` is not found)